### PR TITLE
ath79: Change status LED behavior to ipq40xx style

### PR DESCRIPTION
**Pull request by hanipouspilot**
The unification idea by Mathias Kresin

Signed-off-by: Dmitry Tunin <hanipouspilot@gmail.com>

### CODE REVIEW

**Comment by hanipouspilot** [comment]
@mkresin I added aliases where we had them.

**Comment by hanipouspilot** [comment]
@mkresin Threre is even a better example with 3 leds. I don't know if my choice of statii is the best one.
qca9563_phicomm_k2t.dts

**Comment by mkresin** [comment]
> I don't know if my choice of statii is the best one.

Me neither. But if we have them in an extra commit, we can discuss them and I can cherry pick at least the new diag.sh to not block the other PRs.

Right now, the changes are hidden in a commit which is supposed to change how the leds are addressed but not which leds are used.

**Comment by hanipouspilot** [comment]
OK. I am on it. I'll revert the changes for these two routers, and leave only the DIR-825B1

**Comment by mkresin** [comment]
> I'll revert the changes for these two routers,

Just move them into an extra commit. This way we have something we can discuss about. For the DIR-825B1 changes an extra commit as well. I will cherry pick this one together with the base commit.

**Comment by hanipouspilot** [comment]
@mkresin Done.
Changes to specific routers can be added later.

**Comment by hanipouspilot** [comment]
Forgot to remove DIR825 to an extra commit.

**Comment by hanipouspilot** [comment]
I need to dome some fixes.

**Comment by mkresin** [comment]
@hnyman any opinion about the wndr3700/3800 related commit? Please consider it as RFC. I'm not going to include this commit without your ACK.

**Comment by hnyman** [comment]
I have no objections

But to be honest, I am not quite sure of the value of the color change during the boot process on WNDR3800.

I tested:
* Earlier ar71xx and earlier ath79: first u-boot led action, then early preinit dark, then GREEN failsafe entrance indication rapid, then GREEN normal blinking preinit, then steady green when boot completes
* This PR: first u-boot led action, then early preinit dark, then ORANGE failsafe entrance indication rapid, then ORANGE normal blinking preinit, then steady green when boot completes

If it would first be GREEN during early preinit, then ORANGE during failsafe indication window, then again GREEN during normal late preinit, then there would be some signal value regarding the hard-to-spot (ridiculously short) 2 second failsafe entrance window. (Hmmm. I am not sure if it is even possible to have led action before the preinit entrance window. And there is already now the change from dark to rapid blink. So maybe I am just rambling on)

But like said, no objections.

**Comment by hanipouspilot** [comment]
@hnyman I had completely no idea about how it was supposed to be by design.
It can be done either way.

**Comment by hnyman** [comment]
Nice work.
Regarding WNDR3700/3800: I propose the following changed to DTS:
* "boot" led alias is power_green (not orange)
* power_green defaults state = "on"  (new change)

This makes it behave more like intended by mkresin, i.e. the green led being mainly used, and orange only if already in the failsafe.  (Green led turns on earlier steady, then blinks rapidly during failsafe entry query, and then slower during the remaining boot. If failsafe is entered, then orange led blinks rapidly)

```
--- a/target/linux/ath79/dts/ar7161_netgear_wndr3700.dtsi
+++ b/target/linux/ath79/dts/ar7161_netgear_wndr3700.dtsi
@@ -8,7 +8,10 @@
 
 / {
        aliases {
-               led-status = &power_green;
+               led-boot = &power_green;
+               led-failsafe = &power_orange;
+               led-running = &power_green;
+               led-upgrade = &power_orange;
        };
 
        chosen {
@@ -45,10 +48,10 @@
                power_green: power_green {
                        label = "netgear:green:power";
                        gpios = <&gpio 2 GPIO_ACTIVE_LOW>;
-                       default-state = "off";
+                       default-state = "on";
                };
 
-               power_orange {
+               power_orange: power_orange {
                        label = "netgear:orange:power";
                        gpios = <&gpio 1 GPIO_ACTIVE_LOW>;
                        default-state = "off";


```

File: target/linux/ath79/dts/ar7161_netgear_wndr3700.dtsi [hunk 1]
```diff
@@ -8,10 +8,10 @@
 
 / {
 	aliases {
-		led-boot = &power_green;
-		led-failsafe = &power_green;
+		led-boot = &power_orange;
```
Comment (hnyman) [inline target/linux/ath79/dts/ar7161_netgear_wndr3700.dtsi]: This "boot" should remain power_green.

File: target/linux/ath79/dts/ar7161_netgear_wndr3700.dtsi [hunk 2]
```diff
@@ -51,7 +51,7 @@
 			default-state = "off";
```
Comment (hnyman) [inline target/linux/ath79/dts/ar7161_netgear_wndr3700.dtsi]: power_green default state should be "on"

**Review by hnyman — APPROVED** [review]

**Comment by mkresin** [comment]
@hnyman I worded the mail I send to you badly. The idea is to use the same LED as the bootloader (for the WNDR orange) and only switch to the other LED (here green) if the boot is done.

I'll change the default state before pushing the change. I would like to use the two boards covered by the PR as reference implementations.

**Comment by mkresin** [comment]
merged, thanks!

**Pull request closed without merging**